src/QsCompiler/DataStructures/Diagnostics.fs

```diff
@@ -874,25 +906,21 @@ type DiagnosticItem =
                 "This expression may be short-circuited, and operation calls may not be executed."
             | WarningCode.DeprecationWithRedirect -> "{0} has been deprecated. Please use {1} instead."
             | WarningCode.DeprecationWithoutRedirect -> "{0} has been deprecated."
+
+            // Reuse error messages and hope that the expanded message doesn't contain {}.
             | WarningCode.UnsupportedResultComparison ->
-                "The target {0} does not support comparing measurement results."
+                DiagnosticItem.Message(ErrorCode.UnsupportedResultComparison, args)
```

Comment:
This is similar to how it was done in the original change and is also what I'm stumbled upon. The goal here is to reuse corresponding error message for a warning. We don't have a separate function to get a string without parameters applied. Therefore, we get a string with parameters applied. Later no parameters will be applied because expanded string doesn't have {} (hopefully). I don't like this, just followed the same approach as in the original change that I got as a reference. Since you also raised this question, I will change the code and add a function that would produce unexpanded error strings.

Reply:
Added member to reuse capability error strings.